core/src/test/java/org/elasticsearch/search/aggregations/bucket/InternalSingleBucketAggregationTestCase.java

```diff
@@ -77,4 +81,36 @@ protected final void assertReduced(T reduced, List<T> inputs) {
         }
         extraAssertReduced(reduced, inputs);
     }
+
+    @Override
+    protected void assertFromXContent(T aggregation, ParsedAggregation parsedAggregation) {
+        assertTrue(parsedAggregation instanceof ParsedSingleBucketAggregation);
+        ParsedSingleBucketAggregation parsed = (ParsedSingleBucketAggregation) parsedAggregation;
+
+        assertEquals(aggregation.getDocCount(), parsed.getDocCount());
+        InternalAggregations aggregations = aggregation.getAggregations();
+        Map<String, Aggregation> expectedAggregations = new HashMap<>();
+        int expectedNumberOfAggregations = 0;
+        for (Aggregation expectedAggregation : aggregations) {
+            // since we shuffle xContent, we cannot rely on the order of the original inner aggregations for comparison
+            assertTrue(expectedAggregation instanceof InternalAggregation);
+            expectedAggregations.put(expectedAggregation.getName(), expectedAggregation);
+            expectedNumberOfAggregations++;
+        }
+        int parsedNumberOfAggregations = 0;
+        for (Aggregation parsedAgg : parsed.getAggregations()) {
+            assertTrue(parsedAgg instanceof ParsedAggregation);
+            assertTrue(expectedAggregations.keySet().contains(parsedAgg.getName()));
+            // TODO check aggregation equality, e.g. same xContent
+            parsedNumberOfAggregations++;
+        }
+        assertEquals(expectedNumberOfAggregations, parsedNumberOfAggregations);
+        Class<? extends ParsedSingleBucketAggregation> parsedClass = implementationClass();
+        assertTrue(parsedClass != null && parsedClass.isInstance(parsedAggregation));
+    }
+
+    // TODO make abstract
```

Comment:
Now that all subclasses are actually converted in one PR this can go completely.